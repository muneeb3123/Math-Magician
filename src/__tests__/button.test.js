import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../components/button';

describe('Button', () => {
  it('should render a button with the provided value', () => {
    const value = '5';
    const onClick = jest.fn();

    const { getByText } = render(<Button value={value} onClick={onClick} />);
    const buttonElement = getByText(value);
    expect(buttonElement).toMatchSnapshot();
  });

  it('should call the onClick function when the button is clicked', () => {
    const value = '9';
    const onClick = jest.fn();
    const { getByText } = render(<Button value={value} onClick={onClick} />);

    const buttonElement = getByText(value);
    fireEvent.click(buttonElement);

    expect(onClick).toHaveBeenCalledWith(value);
  });
});
