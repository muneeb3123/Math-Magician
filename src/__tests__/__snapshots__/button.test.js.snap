// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a button with the provided value 1`] = `
<button
  class="silver button-5 calculator-button"
  type="button"
>
  5
</button>
`;
